service/controller/app/v1/resource/namespace/delete.go

```diff
@@ -0,0 +1,21 @@
+package namespace
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
```

Comment:
I think we don't need to take care of `EnsureCreated` case, does it?

Reply:
Yes I don't think we need to implement EnsureCreated.

But we have a naming collision here. 😢 

For bootstrapping chart-operator in tenant clusters we need a namespace resource which will create the giantswarm namespace in new tenant clusters.

Can we rename this to `appnamespace`? Please also add godoc so the purpose of this resource is clear.

Reply:
@rossf7 Can you suggest where you want to see the godoc description? I'm missing in here.